### PR TITLE
feat: preload and push modern resources in modern mode

### LINKED ISSUE #4351: Modern build assets are cached and loaded by nuxt-pwa workbox
### What problem does this feature solve?

I just enabled the "client modern" build and it works great! Good job friends. 

Additionally, I'm running with nuxt-pwa which grabs all files generated in `_nuxt`. This doubles the files loaded by the service worker where half of the files will never be requested due to the client-side "module"/"nomodule" bridge.

### What does the proposed changes look like?

I propose a file naming change of the modern files. If the modern JS assets would would have a different file name (thinking of the prefix `m-` or something) or would live in another directory making the SW load only the modern asset relatively easily.

Happy to propose a PR if you think that makes sense. :)

## Edited

indeed I see that new assets are prefixed in dev mode. :)

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c8157">#c8157</a>)</em></sub></div>

### PR DESCRIPTION
**Pull request by clarkdo**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Related: #4356 
Resolves: #4351 

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [x] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

File: test/unit/modern.server.test.js [hunk 1]
```diff
@@ -37,6 +34,31 @@ describe('modern server mode', () => {
...
+    )
+  })
+
+  test('should contain module http2 pushed resources', async () => {
+    const { headers: { link } } = await rp(url('/'), {
+      headers: { 'user-agent': modernUA },
+      resolveWithFullResponse: true
+    })
+    expect(link).toEqual(
+      `</_nuxt/modern-runtime.js>; rel=preload; as=script${
```
Comment (manniL) [inline test/unit/modern.server.test.js]: Maybe use an array + join here instead of the (clever!) template literal hack?

Comment (clarkdo) [reply test/unit/modern.server.test.js]: OK, I'll change to it.

File: packages/vue-renderer/src/renderer.js [hunk 2]
```diff
@@ -63,14 +63,30 @@ export default class VueRenderer {
     return context.renderScripts()
   }
 
+  getPreloadFiles(context) {
+    const legacyFiles = context.getPreloadFiles()
```
Comment (manniL) [inline packages/vue-renderer/src/renderer.js]: Not sure if "legacy" is the best naming here :thinking: Maybe `transpiled` would fit better?

Comment (manniL) [reply packages/vue-renderer/src/renderer.js]: Oh! Good point, didn't know that ☺️

**Comment by clarkdo** [comment]
@pi0 Could you review this one ? 😄

**Review by TheAlexLichter — DISMISSED** [review]

**Review by aldarund — DISMISSED** [review]

**Review by aldarund — DISMISSED** [review]

**Review by aldarund — APPROVED** [review]

**Pull request merged**